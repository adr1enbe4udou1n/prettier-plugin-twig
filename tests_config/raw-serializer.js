const RAW = Symbol.for("raw");

const print = val => val[RAW];
const test = val =>
    val &&
    Object.prototype.hasOwnProperty.call(val, RAW) &&
    typeof val[RAW] === "string";
export default {
    print,
    test
};
