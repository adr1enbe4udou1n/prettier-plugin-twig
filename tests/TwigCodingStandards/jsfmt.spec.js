import { run_spec } from "tests_config/run_spec";
import { describe, expect, it } from "vitest";

/** @type {import('tests_config/run_spec').PrettierOptions} */
const formatOptions = {
    twigAlwaysBreakObjects: false
};

describe("Twig coding standards", () => {
    it("should follow twig coding standards", async () => {
        const { actual, snapshotFile } = await run_spec(import.meta.url, {
            source: "twigCodingStandards.twig"
        });
        await expect(actual).toMatchFileSnapshot(snapshotFile);
    });
});
