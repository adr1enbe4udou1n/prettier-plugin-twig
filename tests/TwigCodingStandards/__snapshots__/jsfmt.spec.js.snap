// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`twigCodingStandards.twig - twig-verify > twigCodingStandards.twig 1`] = `
{{ foo }}
{# comment #}
{% if foo %}{% endif %}

{{- foo -}}
{#- comment -#}
{%- if foo -%}{%- endif -%}

{{ 1 + 2 }}
{{ foo ~ bar }}
{{ true ? true : false }}

{{ [1, 2, 3] }}
{{ {'foo': 'bar'} }}

{{ 1 + (2 * 3) }}

{{ foo|upper|lower }}
{{ user.name }}
{{ user[name] }}
{% for i in 1..12 %}{% endfor %}

{{ foo|default('foo') }}
{{ range(1..10) }}

{% block foo %}
    {% if true %}
        true
    {% endif %}
{% endblock %}
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
{{ foo }}
{# comment #}
{% if foo %}{% endif %}

{{- foo -}}
{#- comment -#}
{%- if foo -%}{%- endif -%}

{{ 1 + 2 }}
{{ foo ~ bar }}
{{ true ? true : false }}

{{ [1, 2, 3] }}
{{ { foo: 'bar' } }}

{{ 1 + 2 * 3 }}

{{ foo|upper|lower }}
{{ user.name }}
{{ user[name] }}
{% for i in 1..12 %}{% endfor %}

{{ foo|default('foo') }}
{{ range(1..10) }}

{% block foo %}
    {% if true %}
        true
    {% endif %}
{% endblock %}

`;
